pkg/vsphere/tasks/waiter.go

```diff
@@ -60,6 +72,23 @@ func WaitForResult(ctx context.Context, f func(context.Context) (Task, error)) (
 	var info *types.TaskInfo
 	var backoffFactor int64 = 1
 
+	// initialize the limiter channel once
+	once.Do(func() {
+		inFlight = make(chan struct{}, maxInFlight)
+	})
+
+	// reserve the spot
+	inFlight <- struct{}{}
+
+	// check whether we timed out/canceled
+	select {
+	case <-ctx.Done():
+		return nil, ctx.Err()
+	default:
+	}
+	// give the spot back
+	defer func() { <-inFlight }()
```

Comment:
👍  doing that but I think I won't be merging this until we discuss all the possible solutions first